src/sprints/sprints.service.ts

```diff
@@ -273,91 +280,88 @@ export class SprintsService {
+            where: {
+                id: meetingId,
+            },
+            data: {
+                title,
+                description,
+                meetingLink,
+                dateTime,
+                notes,
+            },
+        });
     }
 
     async createMeetingAgenda(
         meetingId: number,
         { title, description, status }: CreateAgendaDto,
+        req: CustomRequest,
     ) {
-        try {
-            const newAgenda = await this.prisma.agenda.create({
-                data: {
-                    teamMeetingId: meetingId,
-                    title,
-                    description,
-                    status,
-                },
-            });
-            return newAgenda;
-        } catch (e) {
-            if (e.code === "P2003") {
-                throw new BadRequestException(
-                    `Invalid meetingId: ${meetingId}`,
-                );
-            }
-        }
+        await manageOwnTeamMeetingOrAgendaById({ user: req.user, meetingId });
+
+        return this.prisma.agenda.create({
+            data: {
+                teamMeetingId: meetingId,
+                title,
+                description,
+                status,
+            },
+        });
     }
 
     async updateMeetingAgenda(
         agendaId: number,
         { title, description, status }: UpdateAgendaDto,
+        req: CustomRequest,
     ) {
-        try {
-            const updatedMeeting = await this.prisma.agenda.update({
-                where: {
-                    id: agendaId,
-                },
-                data: {
-                    title,
-                    description,
-                    status,
-                },
-            });
-            return updatedMeeting;
-        } catch (e) {
-            if (e.code === "P2025") {
-                throw new NotFoundException(`Invalid agendaId: ${agendaId}`);
-            }
-        }
+        await manageOwnTeamMeetingOrAgendaById({ user: req.user, agendaId });
+        return this.prisma.agenda.update({
+            where: {
+                id: agendaId,
+            },
+            data: {
+                title,
+                description,
+                status,
+            },
+        });
     }
 
-    async deleteMeetingAgenda(agendaId: number) {
-        try {
-            return await this.prisma.agenda.delete({
-                where: {
-                    id: agendaId,
-                },
-            });
-        } catch (e) {
-            if (e.code === "P2025") {
-                throw new NotFoundException(
-                    `${e.meta.cause} agendaId: ${agendaId}`,
-                );
-            }
-        }
+    async deleteMeetingAgenda(agendaId: number, req: CustomRequest) {
+        await manageOwnTeamMeetingOrAgendaById({ user: req.user, agendaId });
+
+        return await this.prisma.agenda.delete({
+            where: {
+                id: agendaId,
+            },
+        });
```

Comment:
redundant `await` when return ing the result of `this.prisma.aganda.delete`
```suggestion
        return this.prisma.agenda.delete({
            where: {
                id: agendaId,
            },
        });
```